### PR TITLE
Issue 109 Use HTL comments

### LINKED ISSUE #109: Use HTL comments
If you want to place comments regarding your code, make sure they don't display to end users.
```
<!--/* Good, this comment will not be rendered */-->
 
<!-- Potentially bad, plain HTML comments will be visible on the page -->
```

### PR DESCRIPTION
**Pull request by GulderBone**
fixes #109

### CODE REVIEW

File: src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import org.sonar.check.Rule;
+import org.sonar.plugins.html.node.CommentNode;
+
+@AemVersion(
+    all = true
+)
+@Metadata(
+    technicalDebt = "5min"
+)
+@ParsingErrorRule
```
Comment (chutch) [inline src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java]: What this annotation is for?

File: src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+import com.cognifide.aemrules.htl.api.ParsingErrorRule;
+import com.cognifide.aemrules.metadata.Metadata;
+import com.cognifide.aemrules.tag.Tags;
+import com.cognifide.aemrules.version.AemVersion;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.plugins.html.node.CommentNode;
+
+@AemVersion(
```
Comment (chutch) [inline src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java]: Please keep the order of annotations.

File: src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java (hunk 1 above)
Comment (GulderBone) [reply src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java]: deleted

File: src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java (hunk 2 above)
Comment (GulderBone) [reply src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java]: changed

File: src/test/java/com/cognifide/aemrules/htl/HtlProfileTest.java [hunk 3]
```diff
@@ -42,6 +43,19 @@
...
+    @Test
+    public void sanity() {
+        profile.define(context);
+        Map<String, BuiltInActiveRule> activeRules = getActiveRulesByRuleKey(context);
+
+        assertThat(activeRules.size()).isEqualTo(3);
+
+        assertThat(activeRules.keySet()).contains("HTL-0");
+        assertThat(activeRules.keySet()).contains("HTL-3");
+
```
Comment (michaltobiasz) [inline src/test/java/com/cognifide/aemrules/htl/HtlProfileTest.java]: This line is not needed

File: src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+)
+@AemVersion(
+    all = true
+)
+@Metadata(
+    technicalDebt = "5min"
+)
+public class HtlCommentsCheck extends AbstractHtlCheck {
+
+    static final String RULE_KEY = "HTL-3";
```
Comment (michaltobiasz) [inline src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java]: Please double check if rule is not duplicated.

File: README.md [hunk 5]
```diff
@@ -78,6 +78,9 @@ Below you will find descriptions of all rules available in **AEM Rules for Sonar
 - **AEM-19** Implicit search strategy used in Sling Query
   - `SearchStrategy` can have negative performance impact if mismatched.
   Therefore developer should always make informed decision and define strategy explicitly.
+  
+- **HTL-3** Usage of HTML comments should be avoided if possible
```
Comment (michaltobiasz) [inline README.md]: Move description like in https://github.com/Cognifide/AEM-Rules-for-SonarQube/pull/137/files#diff-04c6e90faac2675aa89e2176d2eec7d8

File: src/test/java/com/cognifide/aemrules/htl/HtlProfileTest.java (hunk 3 above)
Comment (GulderBone) [reply src/test/java/com/cognifide/aemrules/htl/HtlProfileTest.java]: deleted

File: src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java (hunk 4 above)
Comment (GulderBone) [reply src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java]: Checked, changed to HTL-6

File: README.md (hunk 5 above)
Comment (GulderBone) [reply README.md]: moved

File: README.md [hunk 6]
```diff
@@ -55,6 +55,9 @@ gradlew sonarQube -DsonarRunner.aemVersion=6.4
 
 Below you will find descriptions of all rules available in **AEM Rules for SonarQube** plugin.
 
+- **HTL-6** Usage of HTML comments should be avoided if possible
+  - If you want to place comments regarding your code, make sure they don't display to end users.
```
Comment (chutch) [inline README.md]:
```suggestion
  - If you want to place comments regarding your code, make sure they don't display to the end users.
```

**Comment by GulderBone** [comment]
Closing this one as I opened a new one: https://github.com/Cognifide/AEM-Rules-for-SonarQube/pull/157

**Pull request closed without merging**